### PR TITLE
docs: deploy kubesphere on azure vm

### PR DESCRIPTION
**Pull request by RolandMa1986**
docs: deploy kubesphere on azure vm

### CODE REVIEW

File: content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
+---
+title: "Deploy KubeSphere on Azure VM Instance"
+keywords: "Kubesphere, Installation, HA, high availability, load balancer, Azure"
+description: "The tutorial is for installing a high-availability cluster on Azure."
+---
+
+## Before you begin
+
+Technically, you can either install, administer, and manage Kubernetes yourself or go for a managed Kubernetes solution. If you are looking for a way to take advantage of Kubernetes with a hands-off approach, a fully managed platform solution is what you’re looking for, please see [Deploy KubeSphere on AKS](../../../installing-on-kubernetes/hosted-kubernetes/install-ks-on-aks) for more details. But if you want a bit more control over your configuration and setup a Highly Available clusters on Azure, this instruction will help you to setup a production-ready Kubernetes and KubeSphere.
```
Comment (FeynmanZhou) [inline content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md:9]: a Highly Available clusters => a highly available cluster

File: content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+## Prerequisites
+
+- You need an [Azure](https://portal.azure.com) account to create all the resources.
+- Basic knowledge of [Azure Resource Manager](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/)(ARM) templates, which are files that define the Azure infrastructure and configuration.
+- Considering data persistence, for a production environment, we recommend you to prepare persistent storage and create a StorageClass in advance. For development and testing, you can use the integrated OpenEBS to provision LocalPV as the storage service directly.
+
+## Architecture
+
+Six machines of "Ubuntu 18.04" will be deployed in Azure Resources Group. Tree of them are grouped into an Availability sets, which will be used by Master and ETCD. Another tree VMs will be defined as a VMSS, Worker nodes will be run on it.
```
Comment (FeynmanZhou) [inline content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md:28]: Tree? I guess it is `three` here   :)

Comment (FeynmanZhou) [reply content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md:28]: which will be used by Master and ETCD. => playing the role of both Master and ETCD of the Kubernetes control plane.

Comment (FeynmanZhou) [reply content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md:28]: which will be used by Master and ETCD. => playing the role of both Master and ETCD of the Kubernetes control plane.

File: content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+- Fill in the Admin Username.
+- Copy your public ssh key and fill in the Admin Key. Or create new one with *ssh-keygen*.
+
+> Password authentication is restriced in the Linux configuration, only SSH accept.
+
+Click the *Purchase* button in the bottom when you ready to continue.
+
+### Review Azure Resources in the Portal
+
+Once the deployment success, you can find all the resources you need in the KubeSphereVMRG resource group. Take your time and check them one by one if you are new to Azure. Then find the public IP of LB and private IP addresses of the VMs. You need them in the next step.
```
Comment (FeynmanZhou) [inline content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md:69]: You will need them in the next step.

File: content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+![New Created Resources](/images/docs/aks/azure-vm-all-resources.png)
+
+## Deploy Kubernetes and Kubesphere
+
+For the following commands, you can execute them on your laptop or SSH to one of the Master VMs. There are files will be downloaded to local and disturbed to each VM during the installation. The installation will be much faster when you use **kk** in the Intranet than the Internet.
```
Comment (FeynmanZhou) [inline content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md:75]: you can execute the following command on your laptop or SSH to one of the Master VMs.

File: content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+[Kubekey](https://github.com/kubesphere/kubekey) is the next-gen installer which is used for installing Kubernetes and KubeSphere v3.0.0 fastly, flexibly and easily.
+
+1. First, download it and generate a configuration file to customize the installation as follows.
+
+```
+curl -O -k https://kubernetes.pek3b.qingstor.com/tools/kubekey/kk
+chmod +x kk
+```
+
+2. Then create an example configuration file with default configurations. Here we use Kubernetes v1.17.9 as an example.
```
Comment (FeynmanZhou) [inline content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md:96]: You would better to list how many k8s versions are supported by Kubekey.

File: content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+  2. Please modify the default password after login.
+#####################################################
+https://kubesphere.io             2020-xx-xx xx:xx:xx
+```
+
+### Access KubeSphere Console
+
+Now that KubeSphere is installed, you can access the web console of KubeSphere by the following URL.
+
+-  http://{LoadBalancer IP}:30880
```
Comment (FeynmanZhou) [inline content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md:198]: `Now that KubeSphere is installed, you can access the web console of KubeSphere by the following URL.
-  http://{LoadBalancer IP}:30880`

=>
Congratulation! Now you can access the KubeSphere console using `http://10.128.0.44:30880` (Replace the IP with yours).

Comment (FeynmanZhou) [inline content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md:198]: BTW, if you use the type of LoadBalancer to expose the services, why you still add NodePort `30880` here?

**Review by FeynmanZhou — CHANGES_REQUESTED** [review]
See my comments above

File: content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md (hunk 6 above)
Comment (RolandMa1986) [reply content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md:198]: We can either NodePort or ingress

**Pull request merged**